ext/pcre/php_pcre.c

```diff
@@ -704,7 +695,6 @@ PHPAPI pcre_cache_entry* pcre_get_compiled_regex_cache(zend_string *regex)
 			case 'D':	coptions |= PCRE2_DOLLAR_ENDONLY;break;
 			case 'S':	/* Pass. */					break;
 			case 'U':	coptions |= PCRE2_UNGREEDY;		break;
-			case 'X':	extra_coptions &= ~PCRE2_EXTRA_BAD_ESCAPE_IS_LITERAL;			break;
```

Comment:
I wonder if this would still allow the modifier but ignore it? Like for `S`.

Reply:
It doesn't - this is reflected by the updated testcase; see https://github.com/php/php-src/pull/4429/files#diff-22e7527fa4a1643c926c189b7cffd8db